subprojects/docs/src/docs/userguide/basePlugin.adoc

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[base_plugin]]
+== The Base Plugin
+
+The Base Plugin provides features that are common to most builds and adds a structure to the build that promotes consistency in how they are run. Its most significant contribution is a set of <<sec:base_tasks,'lifecycle' tasks>> that act as an umbrella for the more specific tasks provided by other plugins and build authors.
+
+[[sec:base_plugin_usage]]
+=== Usage
+
+----
+plugins {
+    id 'base'
+}
+----
+
+[[sec:base_tasks]]
+=== Tasks
+
+`clean` — type: `Delete`::
+Deletes the build directory and everything in it, i.e. the path specified by the `buildDir` project property.
+
+`check` — _lifecycle task_::
+Plugins and build authors should attach their verification tasks, such as ones that run tests, to this lifecycle task using `check.dependsOn(__task__)`.
+
+`assemble` — _lifecycle task_::
+Plugins and build authors should attach tasks that produce distributions and other consumable artifacts to this lifecycle task. For example, `jar` produces the consumable artifact for Java libraries. Attach tasks to this lifecycle task using `assemble.dependsOn(__task__)`.
+
+`build` — _lifecycle task_::
+(Depends on `assemble` and `check`) Builds everything
```

Comment:
Sentence should end with a dot.

Reply:
I think I wanted to continue that sentence but was distracted. I'll sort it out one way or the other.

 - [ ] Fix this sentence